httpclient.go

```diff
@@ -25,9 +32,60 @@ func DefaultHttpClient() *http.Client {
 		ExpectContinueTimeout: 1 * time.Second,
 	}
 
-	httpClient := &http.Client{
+	return tr, nil
+}
+
+func newHttpClient(tr *http.Transport) *http.Client {
+	return &http.Client{
 		Transport: tr,
 		Timeout:   DefaultHttpTimeout,
 	}
-	return httpClient
+}
+
+// DefaultHttpClient will set up a basic client
+// with default timeouts/proxies/etc.
+func DefaultHttpClient() *http.Client {
+	tr, err := newHttpTransport("")
+	if err != nil {
+		log.Fatal().Err(err).Msg("unexpectedly failed to create a default HttpClient")
+	}
+
+	return newHttpClient(tr)
+}
+
+type HttpClientOpts struct {
+	// Proxy is the string representation of the proxy the client
+	// should use for connections.
+	Proxy string
+}
+
+func HttpClient(opts *HttpClientOpts) (*http.Client, error) {
```

Comment:
should this be `NewHttpClient`?

Reply:
I followed the naming scheme from the existing `DefaultHttpClient()` https://github.com/mondoohq/ranger-rpc/blob/main/httpclient.go#L15 , but this is easy enough to change. Just let me know if I should only change this to NewHttpClient() or also change to NewDefaultHttpClient(), or some combination of these options.